### PR TITLE
Add interaction-based pagination

### PR DESCRIPTION
**Pull request by VelvetThePanda**
# Summary
@kuylar complained that there wasn't pagination for interactions.

# Details
Adds support for interaction-based pagination. In testing as of writing this PR.

Automatically cleans up after 15 minutes after the last interaction, too.

### CODE REVIEW

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 1]
```diff
@@ -625,6 +625,39 @@ public async Task<ReadOnlyCollection<Reaction>> CollectReactionsAsync(DiscordMes
...
+            var bhv = behaviour ?? this.Config.PaginationBehaviour;
+            var del = deletion ?? this.Config.ButtonBehavior;
+            var bts = buttons ?? this.Config.PaginationButtons;
+
+            var builder = new DiscordInteractionResponseBuilder()
+                .WithContent(pages.First().Content)
+                .AddEmbed(pages.First().Embed)
+                .AsEphemeral(ephemeral);
+
+            await interaction.CreateResponseAsync(InteractionResponseType.ChannelMessageWithSource, builder);
```
Comment (kuylar) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:653]: throws 404 if we create a DeferredChannelMessageWithSource response before calling this

*sorry*

File: DSharpPlus.Interactivity/InteractivityExtension.cs [hunk 2]
```diff
@@ -625,6 +625,39 @@ public async Task<ReadOnlyCollection<Reaction>> CollectReactionsAsync(DiscordMes
...
         }
 
+        /// <summary>
+        /// Sends a paginated message in response to an interaction.
+        /// <para>
+        /// <b>Pass the interaction directly. Interactivity will ACK it.</b>
+        /// </para>
+        /// </summary>
+        /// <param name="interaction">The interaction to create a response to.</param>
+        /// <param name="ephemeral">Whether the response should be ephermaral.</param>
```
Comment (kuylar) [inline DSharpPlus.Interactivity/InteractivityExtension.cs:635]:
```suggestion
        /// <param name="ephemeral">Whether the response should be ephemeral.</param>
```

Comment (Giggitybyte) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:635]: 🧀

File: DSharpPlus.Interactivity/InteractivityExtension.cs (hunk 1 above)
Comment (VelvetThePanda) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:653]: It's very clearly stated in the documentation.

Comment (kuylar) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:653]: sorry, didnt see it, but shouldnt there be a way to *not* ack it?

Comment (VelvetThePanda) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:653]: What sense does that make

Comment (VelvetThePanda) [reply DSharpPlus.Interactivity/InteractivityExtension.cs:653]: The ability to make it ack requires making this method far clunkier

File: DSharpPlus.Interactivity/Extensions/InteractionExtensions.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+using System.Threading.Tasks;
+using DSharpPlus.Entities;
+using DSharpPlus.Interactivity.Enums;
+using DSharpPlus.Interactivity.EventHandling;
+
+namespace DSharpPlus.Interactivity.Extensions
+{
+    public static class InteractionExtensions
+    {
+        public static Task SendPaginatedMessageAsync(this DiscordInteraction interaction, bool ephemeral, DiscordUser user, IEnumerable<Page> pages, PaginationButtons? buttons = null, PaginationBehaviour? behaviour = default, ButtonPaginationBehavior? deletion = default, CancellationToken token = default)
```
Comment (IDoEverything) [inline DSharpPlus.Interactivity/Extensions/InteractionExtensions.cs:34]: Could this possibly be named `SendPaginatedResponseAsync`?

**Review by IDoEverything — APPROVED** [review]

**Pull request merged**